tests/expectations/tests/nsBaseHashtable.rs

```diff
@@ -32,6 +30,14 @@ impl Default for nsBaseHashtable_LookupResult {
         unsafe { ::std::mem::zeroed() }
     }
 }
+impl nsBaseHashtable_LookupResult {
+    #[inline]
+    pub unsafe fn new(arg1: *mut nsBaseHashtable_EntryType, arg2: *mut nsBaseHashtable) -> Self {
+        let mut __bindgen_tmp = ::std::mem::uninitialized();
+        nsBaseHashtable_LookupResult_LookupResult(&mut __bindgen_tmp, arg1, arg2);
```

Comment:
Test building is failing due to this line, it looks weird the repeated `LookupResult_LookupResult` string here.

Reply:
Oh I see what's going on... We're now generating constructors and such for structs that don't use any template parameter. That's bogus actually, since `LookupResult` is parameterized by a bunch of stuff, so that condition needs to use `all_template_parameters` or something of the sort.

Reply:
Do you have any advice on where to start looking to solve this problem? I'm kind of ~~completely~~ lost right now.

Reply:
Yes, sorry :)

So this is probably related to this check:

https://github.com/rust-lang-nursery/rust-bindgen/pull/1245/files#diff-567e947a8cc5ad306d3eff99dfb66cc9R1735

Your patch somehow made that condition change. Those are nested template structs, so it's kind of tricky, but I'd double-check with and without your changes running some of the reduced tests.

Reply:
I added manually the extern definitions and now the code compiles. Is this something that should be generated automatically? 

```rust
extern "C" {
    pub fn nsBaseHashtable_EntryPtr_EntryPtr_destructor(this: *mut nsBaseHashtable_EntryPtr);
    pub fn nsBaseHashtable_EntryPtr_EntryPtr(this: *mut nsBaseHashtable_EntryPtr,
                                             arg1: *mut nsBaseHashtable,
                                             arg2: *mut nsBaseHashtable_EntryType,
                                             arg3: bool);
    pub fn nsBaseHashtable_LookupResult_LookupResult(this: *mut nsBaseHashtable_LookupResult,
                                                     arg1: *mut nsBaseHashtable_EntryType,
                                                     arg2: *mut nsBaseHashtable);
}
```

Reply:
Not really, that will compile, but will not link. We can't generate them because they have no symbols, being them in a templated struct.